Assets/MixedRealityToolkit/UX/Scripts/Progress/FadeObject.cs

```diff
@@ -0,0 +1,118 @@
+﻿using UnityEngine;
+
+
+namespace MixedRealityToolkit.Examples.InteractiveElements
+{
+    public class FadeObject : MonoBehaviour
+    {
+        public float FadeTime = 0.5f;
+        public bool AutoFadeIn = false;
+
+        private float mFadeCounter = 0;
+        private Color mCachedColor;
+        private bool mFadingIn = true;
+        private Renderer mRenderer;
+        private bool mIsFading = false;
+
+        // Use this for initialization
+        private void Awake()
+        {
+            mRenderer = this.GetComponent<Renderer>();
+            mCachedColor = mRenderer.material.color;
```

Comment:
Memory leak in this class when calling renderer.material (here and below).

Please properly cache the material and dispose of and created instances.